### ISSUE TITLE
kbox provision fails on Ubuntu 14.04

### ISSUE THREAD

**Issue by badri** [kalabox/kalabox#186]
I tried the following steps in a fresh digitalocean ubuntu 14.04 setup:
1. install nodejs/npm
2. login as non root user.
3. run
```
npm install kalabox -g
kbox provision -- -v
```

I get EACCESS for boot2docker.
```
debug: Finished core-downloads (50960)
info: -- OK! 45% complete! --

debug: [10:20] Starting syncthing-setup
debug: description => Setting up syncthing...
debug: dependencies => core-downloads
info: -- Step 10/20 --
info: Setting up syncthing...
debug: Finished syncthing-setup (910)
info: -- OK! 50% complete! --

debug: [11:20] Starting engine-docker-provider-profile
debug: description => Setting up the provider profile...
debug: dependencies => core-downloads
info: -- Step 11/20 --
info: Setting up the provider profile...
debug: Finished engine-docker-provider-profile (6)
info: -- OK! 55% complete! --

debug: [12:20] Starting services-kalabox-is-dns-set
debug: description => Checking if dns is set...
debug: dependencies => engine-docker-provider-profile
info: -- Step 12/20 --
info: Checking if dns is set...
debug: DNS is not set.
debug: Finished services-kalabox-is-dns-set (88)
info: -- OK! 60% complete! --

debug: [13:20] Starting services-kalabox-install-dns-deps
debug: description => Queuing up services admin commands...
debug: dependencies => services-kalabox-is-dns-set
info: -- Step 13/20 --
info: Queuing up services admin commands...
debug: Finished services-kalabox-install-dns-deps (3)
info: -- OK! 65% complete! --

debug: [14:20] Starting engine-docker-install-engine
debug: description => Queuing provider admin commands...
debug: dependencies => core-auth
info: -- Step 14/20 --
info: Queuing provider admin commands...
debug: Finished engine-docker-install-engine (4)
info: -- OK! 70% complete! --

debug: [15:20] Starting core-run-admin-commands
debug: description => Running admin install commands...
debug: dependencies => engine-docker-install-engine, services-kalabox-install-dns-deps, core-prepare-usr-bin
info: -- Step 15/20 --
info: Running admin install commands...
EACCES, open '/usr/local/bin/boot2docker'
Error: EACCES, open '/usr/local/bin/boot2docker'
[sudo] password for lakshminp:
```

Am I missing any step?
Does it make sense to have boot2docker step at all in Linux? Can't the provider and engine be the same(docker) for Linux envs?

**Comment by pirog** [comment]
As of now kalabox is definitely not meant to be run in this way... ie on a server/cloud out there somewhere. It's intended use case is for local development only. 

That said it looks like you just don't have perms for `/usr/local/bin/boot2docker` set correctly. 

If you wanted to try to get it to work in this way you would want to do exactly what you say, ie run docker straight up instead of in B2D. You could do this by writing a different engine backend for kalabox.

**Comment by badri** [comment]
yes, I tried running on a fresh DO box just to confirm if I was able to reproduce the issue. I had the same problem in my local box too. Thanks for the quick response :)

**Comment by pirog** [comment]
got it!

i would def make sure the b2d is executable.

On Tue, May 26, 2015 at 10:09 PM, Lakshminp <notifications@github.com>
wrote:

> yes, I tried running on a fresh DO box just to confirm if I was able to
> reproduce the issue. I had the same problem in my local box too. Thanks for
> the quick response :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kalabox/kalabox/issues/186#issuecomment-105758973>.
>



-- 
Cheers,

Mike Pirog
Kalamuna
www.kalamuna.com

**Comment by pirog** [comment]
hey @badri 

We just released 0.8.0 which may resolve this issue. I would try uninstalling (https://github.com/kalabox/kalabox/wiki/Uninstalling-Kalabox) pulling in the new code and provisioning using 

```
npm install kalabox -g
kbox provision
```

If you did the developer install you should consult https://github.com/kalabox/kalabox/wiki/Updating-Kalabox
